File: js/generic_tree.js
var readline = require('readline-sync')
class Node{
	constructor(val,child){
		this.val=val
		this.child=child
	}
}

class generic_tree{
	constructor(){
		this.root=null
		this.root=this.add(this.root)
	}
	add(parent){
		let n = new Node(null,[])
		if(parent==null || parent==undefined){
			console.log('enter value\n')
			let value=readline.question()
			n.val=value
			parent=n
		}
		console.log('how many children??')
		let num = readline.question()

		for(let i=0;i<num;i++)
			parent.child.push(this.add(parent.child[i]))
		return parent
	}

	get_root(){
		return this.root
	}

	display(node){
		if(node==null || node==undefined)
			return
		console.log(node.val)
		for(let i=0;i<node.child.length;i++)
			this.display(node.child[i])
	}
}

tree = new generic_tree()
tree.display(tree.get_root())

File: cpp/generic_tree.cpp
#include<bits/stdc++.h>
using namespace std;

struct Node
{
	int val;
	vector<Node*> v;
};
Node* root = NULL;

Node* add(Node* parent){
	if(parent == NULL){
		Node*nn= new Node();
		cout<<"enter the value:\n";
		int value;
		cin>>value;
		nn->val = value;
		parent = nn;
	}

	int n;
	cout<<"how many children??\n";
	cin>>n;
	for(int i=0;i<n;i++){
		parent->v.push_back(NULL);
		parent->v[i] = add(parent->v[i]);
	}

	return parent;
	
}

void display(Node* parent){
	if(parent == NULL)
		return;
	cout<<parent->val<<endl;
	for(int i=0;i<parent->v.size();i++)
		display(parent->v[i]);
}
int main(int argc, char const *argv[])
{
	root = add(root);
	display(root);
	return 0;
}